src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AdjudicationMappingResourceIntTest.kt

```diff
@@ -291,6 +342,260 @@ class AdjudicationMappingResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @DisplayName("POST /mapping/adjudications/all")
+  @Nested
+  inner class CreateAllMappingTest {
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.post().uri("/mapping/adjudications/all")
+        .body(BodyInserters.fromValue(createAllMapping()))
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+      webTestClient.post().uri("/mapping/adjudications/all")
+        .headers(setAuthorisation(roles = listOf()))
+        .body(BodyInserters.fromValue(createAllMapping()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `create visit forbidden with wrong role`() {
+      webTestClient.post().uri("/mapping/adjudications/all")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .body(BodyInserters.fromValue(createAllMapping()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `will create mapping when no hearings or punishments`() = runBlocking {
+      webTestClient.post().uri("/mapping/adjudications/all")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ADJUDICATIONS")))
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(
+          BodyInserters.fromValue(
+            // language=json
+            """{
+              "adjudicationId": {
+                "adjudicationNumber" : $ADJUDICATION_NUMBER,
+                "chargeSequence" : $CHARGE_SEQ,
+                "chargeNumber": "$CHARGE_NUMBER"
+              },
+              "hearings": [],
+              "punishments": [],
+              "label": "2023-04-20",
+              "mappingType": "MIGRATED"
+          }""",
+          ),
+        )
+        .exchange()
+        .expectStatus().isCreated
+
+      val createdMappingByNomisId = webTestClient.get()
+        .uri("/mapping/adjudications/adjudication-number/$ADJUDICATION_NUMBER/charge-sequence/$CHARGE_SEQ")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_ADJUDICATIONS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody(AdjudicationMappingDto::class.java)
+        .returnResult().responseBody!!
```

Comment:
It's unusual for you not to use `jsonPath` - any reason?

Reply:
probably a cut an paste of the original test. So no "good" reason